src/open_inwoner/accounts/tests/test_auth_2fa_sms.py

```diff
@@ -0,0 +1,542 @@
+
+        self.assertFalse(mock_gateway_send.called)
+
+        response = self.client.post(
+            reverse("login"),
+            data={
+                "next": "",
+                "username": self.user.email,
+                "password": "secret",
+            },
+            follow=True,
+        )
+
+        params = {
+            "next": "",
+            "user": signer.sign(self.user.pk),
+        }
+
+        self.assertRedirects(
+            response, reverse("verify_token") + "?" + urlencode(params)
+        )
+        self.assertContains(response, "Account verificatie")
+        self.assertTrue(mock_gateway_send.called)
+        self.assertEqual(mock_gateway_send.call_count, 1)
+        self.assertEqual(mock_gateway_send.call_args[1]["to"], self.user.phonenumber)
+
+        location = reverse("verify_token") + "?" + urlencode(params)
+        response = self.client.post(
+            reverse("resend_token") + "?" + urlencode(params),
+            follow=True,
+            HTTP_REFERER=location,
+        )
+
+        self.assertEqual(mock_gateway_send.call_count, 2)
+        self.assertEqual(mock_gateway_send.call_args[1]["to"], self.user.phonenumber)
+        self.assertRedirects(
+            response, reverse("verify_token") + "?" + urlencode(params)
+        )
+
+        sent_token = mock_gateway_send.call_args[1]["token"]
+
+        response = self.client.post(
+            response._request.get_full_path(),
+            follow=True,
+            data={
+                "token": sent_token,
+            },
+        )
+        self.assertRedirects(response, reverse("pages-root"))
+
+    @patch("open_inwoner.accounts.gateways.gateway.send")
+    def test_login_resend_sms_max_attempts(self, mock_gateway_send):
+        with freeze_time("2023-05-22 12:00:00"):
+            response = self.client.get(reverse("login"))
+
+            self.assertFalse(mock_gateway_send.called)
+
+            response = self.client.post(
+                reverse("login"),
+                data={
+                    "next": "",
+                    "username": self.user.email,
+                    "password": "secret",
+                },
+                follow=True,
+            )
+
+            params = {
+                "next": "",
+                "user": signer.sign(self.user.pk),
+            }
+            self.assertRedirects(
+                response, f"{reverse('verify_token')}?{urlencode(params)}"
+            )
+            self.assertContains(response, "Account verificatie")
+            self.assertTrue(mock_gateway_send.called)
+            self.assertEqual(mock_gateway_send.call_count, 1)
+            self.assertEqual(
+                mock_gateway_send.call_args[1]["to"], self.user.phonenumber
+            )
+
+            location = reverse("verify_token") + "?" + urlencode(params)
+
+            # we have a limit of 25 visit times per 5 minutes
+            for login_attempt in range(1, 26):
```

Comment:
Maybe just lower the limit for the test instead of just hammering through 25 tries :laughing: